test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -0,0 +1,136 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Testing;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+#if NETCOREAPP2_1
+using TestApp.AspNetCore._2._1;
+#elif NETCOREAPP3_1
+using TestApp.AspNetCore._3._1;
+#else
+using TestApp.AspNetCore._5._0;
+#endif
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.AspNetCore.Tests
+{
+    public class MetricTests
+        : IClassFixture<WebApplicationFactory<Startup>>, IDisposable
+    {
+        private readonly WebApplicationFactory<Startup> factory;
+        private MeterProvider meterProvider = null;
+
+        public MetricTests(WebApplicationFactory<Startup> factory)
+        {
+            this.factory = factory;
+        }
+
+        [Fact]
+        public void AddAspNetCoreInstrumentation_BadArgs()
+        {
+            MeterProviderBuilder builder = null;
+            Assert.Throws<ArgumentNullException>(() => builder.AddAspNetCoreInstrumentation());
+        }
+
+        [Fact]
+        public async Task RequestMetricIsCaptured()
+        {
+            var metricItems = new List<MetricItem>();
+            var metricExporter = new TestExporter<MetricItem>(ProcessExport);
+
+            void ProcessExport(Batch<MetricItem> batch)
+            {
+                foreach (var metricItem in batch)
+                {
+                    // The exporter is invoked even when there are no metrics
+                    if (metricItem.Metrics.Count > 0)
+                    {
+                        metricItems.Add(metricItem);
+                    }
+                }
+            }
+
+            this.meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddAspNetCoreInstrumentation()
+                .AddMetricProcessor(new PushMetricProcessor(exporter: metricExporter, exportIntervalMs: 10, isDelta: true))
+                .Build();
+
+            using (var client = this.factory.CreateClient())
+            {
+                var response = await client.GetAsync("/api/values");
+                response.EnsureSuccessStatusCode();
+            }
+
+            // Wait for at least two exporter invocations
+            WaitForMetricItems(metricItems, 2);
+
+            this.meterProvider.Dispose();
+
+            // There should be more than one result here since we waited for at least two exporter invocations.
+            // Is this what we should expect?
+            // Since we've configured delta aggregation does it make sense to continue to export zero values?
```

Comment:
Curious what folks think here. If a metric value is effectively unchanged should we continue to export it? Does the answer differ based on push vs. pull or cumulative vs. delta?

Reply:
I think it is up the the exporter/backend.

Reply:
the sdk can sent null to indicate no new update to this metric since last collect/export. (we don't do this now. something which can be fixed relatively easily) The exporter can decide to export nulls to let its backend knows explicitly that no new update. Or the exporter can chose to drop it.